### PR TITLE
WIP Import Profiles

### PR DESCRIPTION
**Pull request by Yiroh**
Managed to get the software to compile and run, can't test it since I have no CSVs to test with. Submitting pull request for that purpose.

### CODE REVIEW

File: src-core/wealthfolio.db
Comment (afadil) [inline src-core/wealthfolio.db:1]: remove this db.file

File: src-core/migrations/2024-10-15-173026_create_csv_import_profiles/up.sql
Comment (afadil) [inline src-core/migrations/2024-10-15-173026_create_csv_import_profiles/up.sql:1]: I suggest we use only one table with mapping as map stored in text columns. The app will read and parse the mapping from each column. Something like this:

CREATE TABLE IF NOT EXISTS import_mappings (
    account_id TEXT PRIMARY KEY,
    fields_mappings TEXT NOT NULL,
    activity_type_mappings TEXT NOT NULL,
    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
    updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
);

File: src-core/src/db_functions.rs
Comment (afadil) [inline src-core/src/db_functions.rs:1]: move this code to a account_service. with only 2 functions, get_account_import_mapping(account_id) and save_account_import_mapping(account_id, import_mapping)

File: src-core/src/models.rs
Comment (afadil) [inline src-core/src/models.rs:1]: I suggest we keep the model simple:
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ImportMapping {
    pub account_id: String,
    pub fields_mappings: HashMap<String, String>,
    pub activity_type_mappings: HashMap<String, String>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

File: src/commands/activity.ts
Comment (afadil) [inline src/commands/activity.ts:1]: can you please remove all the frontend code, because I completely changed the UX

**Review by afadil — CHANGES_REQUESTED** [review]
Hi,
I left you some comment so the PR can match the new UX for import I'm working on. If you can please make the adjustments.

Thank you so much.

**Comment by Yiroh** [comment]
Made the changes you suggested and simplified an area where it felt more complex than it needed to be (my creation of csv_profile.rs). I wrote an excerpt about it in the paragraph of my commit changes.

**Pull request merged**